tensorboard/webapp/hparams/_redux/hparams_selectors.ts

```diff
@@ -25,81 +30,126 @@ import {
+    ...(otherFilter?.hparams ?? []),
+  ]);
+}
 
-    for (const experimentId of experimentIds) {
-      if (!state.specs[experimentId]) {
-        continue;
-      }
+export const getHparamFilterMap = createSelector(
+  getHparamsState,
+  getHparamsDefaultFiltersForExperiments,
+  getHparamFilterMapResultFunction
+);
 
-      defaultFilterMaps.push(state.specs[experimentId].metric.defaultFilters);
+export function getHparamFilterMapFromExperimentIds(experimentIds: string[]) {
+  return createSelector(
+    getHparamsState,
+    getHparamsDefaultFiltersForExperimentsFromExperimentIds(experimentIds),
+    (hparamState, combinedDefaultFilterMap) =>
+      getHparamFilterMapResultFunction(
+        hparamState,
+        combinedDefaultFilterMap,
+        experimentIds
+      )
+  );
+}
+
+function getMetricsDefaultFiltersForExperimentsResultFunction(
+  state: HparamsState,
+  experimentIds: string[]
+): Map<string, IntervalFilter> {
+  const defaultFilterMaps: Array<Map<string, IntervalFilter>> = [];
+
+  for (const experimentId of experimentIds) {
+    if (!state.specs[experimentId]) {
+      continue;
     }
 
-    return combineDefaultMetricFilters(defaultFilterMaps);
+    defaultFilterMaps.push(state.specs[experimentId].metric.defaultFilters);
   }
+
+  return combineDefaultMetricFilters(defaultFilterMaps);
+}
+
+const getMetricsDefaultFiltersForExperiments = createSelector(
+  getHparamsState,
+  getMetricsDefaultFiltersForExperimentsResultFunction
 );
 
+function getMetricsDefaultFiltersForExperimentsFromExperimentIds(
+  experimentIds: string[]
+) {
+  return createSelector(getHparamsState, (state) =>
+    getMetricsDefaultFiltersForExperimentsResultFunction(state, experimentIds)
+  );
+}
+
+function getMetricFilterMapResultFunction(
+  hparamState: HparamsState,
+  defaultFilterMap: Map<string, IntervalFilter>,
+  experimentIds: string[]
+): Map<string, IntervalFilter> {
+  const id = getIdFromExperimentIds(experimentIds);
+  const otherFilter = hparamState.filters[id];
+
+  return new Map([...defaultFilterMap, ...(otherFilter?.metrics ?? [])]);
+}
+
 export const getMetricFilterMap = createSelector(
-  getMetricsDefaultFiltersForExperiments,
   getHparamsState,
-  (
-    defaultfilterMap,
-    hparamState,
-    experimentIds: string[]
-  ): Map<string, IntervalFilter> => {
-    const id = getIdFromExperimentIds(experimentIds);
-    const otherFilter = hparamState.filters[id];
-
-    return new Map([...defaultfilterMap, ...(otherFilter?.metrics ?? [])]);
-  }
+  getMetricsDefaultFiltersForExperiments,
+  getMetricFilterMapResultFunction
 );
 
+export function getMetricFilterMapFromExperimentIds(experimentIds: string[]) {
+  return createSelector(
+    getHparamsState,
+    getMetricsDefaultFiltersForExperimentsFromExperimentIds(experimentIds),
+    (state, defaultFilterMap) =>
+      getMetricFilterMapResultFunction(state, defaultFilterMap, experimentIds)
+  ) as MemoizedSelector<State, Map<string, IntervalFilter>>;
```

Comment:
What is the purpose of this typecast?

Reply:
I was playing with the typing there for a while, I think this can actually be removed now